exp/zapslog/slog.go

```diff
@@ -25,10 +25,46 @@ import (
 	"runtime"
 
 	"go.uber.org/zap"
+	"go.uber.org/zap/internal/pool"
 	"go.uber.org/zap/zapcore"
 	"golang.org/x/exp/slog"
 )
 
+type fieldSlice []zapcore.Field
+
+func (s *fieldSlice) Append(f zapcore.Field) {
+	*s = append(*s, f)
+}
+
+func (s *fieldSlice) AddTo(enc zapcore.ObjectEncoder) {
+	for i := 0; i < s.Len(); i++ {
+		(*s)[i].AddTo(enc)
+	}
+}
+
+func (s *fieldSlice) Extend(f *fieldSlice) {
+	*s = append(*s, *f...)
+}
+
+func (s *fieldSlice) Len() int {
+	return len(*s)
+}
+
+func (s *fieldSlice) Reset() {
+	*s = (*s)[:0]
+}
+
+func (s *fieldSlice) Return() {
+	s.Reset()
+	_fieldSlices.Put(s)
+}
+
+var _fieldSlices = pool.New(func() *fieldSlice {
```

Comment:
Consider adding a clarifying comment that the pooled value MUST be a pointer type to avoid allocations.